Prettify Sequence diagrams

mr-mocap opened the pull request:
I've only really touched the Sequence diagrams.

Added a UNICODE filled arrowhead (both left & right), A vertical bar with either a left or right fork, and updated the 'dashed' character(s) for a little bit of a prettier look.

Also modified Screen::ASCIIfy() to include the newly added UNICODE chars and substitutions.  Changed ASCIIfication of the vertical-t to just a plain vertical bar, matching the same done with the horizontal-t being replaced with a minus sign.

Added dotted lines to the last two examples (run "diagon Sequence --examples" to see).

## Some Images
UNICODE (in VSCODE in WSL2)
![Screenshot 2025-05-13 100059](https://github.com/user-attachments/assets/9d692622-96d1-44c3-b064-c9b7b875cc21)

ASCII
![Screenshot 2025-05-11 175431](https://github.com/user-attachments/assets/95423550-b9a8-49f3-8c8b-88f2f9b9108a)

![Screenshot 2025-05-11 183958](https://github.com/user-attachments/assets/939dae46-ec2b-4a6a-8794-e1a37d0631ed)

mr-mocap committed:
Prettify Sequence diagrams

mr-mocap committed:
Make dashes twice as long (the longest)
Don't substiture dashed corners, since there aren't any to draw.

mr-mocap committed:
Forgot to add style 1 transformations

mr-mocap committed:
Modify examples to include both plain & dashed message lines

mr-mocap committed:
Change vertical t's to vertical dotted line only
This seems to look better and more consistent because the vertical-t breaks up the dotted vertical line.

ArthurSonzogni commented:
Thanks! I am working in this review. I am also traveling at the moment, so please expect some delay.

ArthurSonzogni committed:
Update expectations, add example, CHANGELOG.

- Update tests expectations.
- Add an entry to the CHANGELOG.
- Revert change to examples.
- Add dashed arrow examples.
- Add an entry to the CHANGELOG.
- Revert changes to the lines.

ArthurSonzogni committed:
Comment code in tests.

ArthurSonzogni committed:
Add tests for Sequence_ascii_only=true

ArthurSonzogni committed:
Fix asciify.

mr-mocap commented:
Apologies for leaving a bit too much in.  I was hoping you would see the direction I was wanting to go with this, and decided to leave it in for you to decide.

Another part, Message::Draw() (beginning), I forgot about because I've made so many experiments.

Perhaps I should open a Discussion for these other improvements instead.
